src/main/java/com/github/cameltooling/lsp/internal/references/ReferencesProcessor.java

```diff
@@ -99,16 +100,34 @@ private boolean isReference(CamelURIInstance camelURIInstanceToSearchReference,
 		return allCamelUriInstance;
 	}
 	
-	private String getDirectId(CamelURIInstance camelDirectURIInstance) {
+	private static String getDirectId(CamelURIInstance camelDirectURIInstance) {
 		Set<PathParamURIInstance> pathParams = camelDirectURIInstance.getComponentAndPathUriElementInstance().getPathParams();
 		if (!pathParams.isEmpty()) {
 			return pathParams.iterator().next().getValue();
 		}
 		return null;
 	}
 
-	private boolean isDirectComponentKind(CamelURIInstance camelURIInstanceToSearchReference) {
+	public static boolean isDirectComponentKind(CamelUriElementInstance camelURIInstanceToSearchReference) {
 		return POSSIBLE_DIRECT_REFERENCE.contains(camelURIInstanceToSearchReference.getComponentName());
 	}
 
+	public static List<String> retrieveEndpointIDsOfScheme(String scheme, ParserXMLFileHelper xmlFileHelper, TextDocumentItem docItem) throws Exception {
+		List<Node> allEndpoints = xmlFileHelper.getAllEndpoints(docItem);
+		List<String> endpointIDs = new ArrayList<>();
+		for (Node endpoint : allEndpoints) {
+			String uriToParse = CamelXmlHelper.getSafeAttribute(endpoint, "uri");
+			if (uriToParse != null) {
+				CamelURIInstance uriInstance = new CamelURIInstance(uriToParse, endpoint, docItem);
+				if (isDirectComponentKind(uriInstance) && uriInstance.getComponentName().equalsIgnoreCase(scheme)) {
+					String dId = getDirectId(uriInstance);
+					String directValue = String.format("%s:%s", scheme, dId);
```

Comment:
I don't understand why there is a String.format("%s:%s").
Seems like connection between direct and direct-vm won't be handled

Reply:
I am still wondering if you can wire direct and direct-vm. Documentation is a bit sparse on that topic.